src/drivers/mediatek/mt8196/ipc.c

```diff
@@ -127,6 +127,10 @@ int platform_ipc_init(struct ipc *ipc)
 	struct ipc_data *iipc;
 
 	iipc = rzalloc(SOF_MEM_FLAG_KERNEL, sizeof(*iipc));
+	if (!iipc) {
+		tr_err(&ipc_tr, "Unable to allocate memory for IPC data");
+		return -ENOMEM;
+	}
```

Comment:
this function panics on all other failures, should it panic here too?

Reply:
I've made it panic for platforms with no MMU atm, but long term this will become userspace so safe to return an error later on as userspace is added.